addons/portal/models/mail_message.py

```diff
@@ -72,11 +73,19 @@ def _portal_message_format(self, properties_names, options=None):
         if 'attachment_ids' in properties_names:
             properties_names.remove('attachment_ids')
             attachments_sudo = self.sudo().attachment_ids
-            attachments_sudo.generate_access_token()
             related_attachments = {
-                att_read_values['id']: att_read_values
-                for att_read_values in attachments_sudo.read(
-                    ["access_token", "checksum", "id", "mimetype", "name", "res_id", "res_model"]
+                att_read_values["id"]: {
+                    **att_read_values,
+                    "as_author_access_token": limited_field_access_token(
+                        att, "as_author_access_token"
+                    ),
```

Comment:
Similar as in to store, this should only be given if the current user is author of the message.